File: bindings/lua/Polycode/GPUDrawBuffer.lua
class "GPUDrawBuffer"


function GPUDrawBuffer:__getvar(name)
	if name == "targetFramebuffer" then
		local retVal = Polycode.GPUDrawBuffer_get_targetFramebuffer(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["RenderBuffer"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "projectionMatrix" then
		local retVal = Polycode.GPUDrawBuffer_get_projectionMatrix(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Matrix4"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "viewMatrix" then
		local retVal = Polycode.GPUDrawBuffer_get_viewMatrix(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Matrix4"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "cameraMatrix" then
		local retVal = Polycode.GPUDrawBuffer_get_cameraMatrix(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Matrix4"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "clearColor" then
		local retVal = Polycode.GPUDrawBuffer_get_clearColor(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Color"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "clearDepthBuffer" then
		return Polycode.GPUDrawBuffer_get_clearDepthBuffer(self.__ptr)
	elseif name == "clearColorBuffer" then
		return Polycode.GPUDrawBuffer_get_clearColorBuffer(self.__ptr)
	elseif name == "backingResolutionScale" then
		local retVal = Polycode.GPUDrawBuffer_get_backingResolutionScale(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector2"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "globalMaterial" then
		local retVal = Polycode.GPUDrawBuffer_get_globalMaterial(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Material"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "viewport" then
		local retVal = Polycode.GPUDrawBuffer_get_viewport(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Rectangle"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
end

function GPUDrawBuffer:__setvar(name,value)
	if name == "targetFramebuffer" then
		Polycode.GPUDrawBuffer_set_targetFramebuffer(self.__ptr, value.__ptr)
		return true
	elseif name == "projectionMatrix" then
		Polycode.GPUDrawBuffer_set_projectionMatrix(self.__ptr, value.__ptr)
		return true
	elseif name == "viewMatrix" then
		Polycode.GPUDrawBuffer_set_viewMatrix(self.__ptr, value.__ptr)
		return true
	elseif name == "cameraMatrix" then
		Polycode.GPUDrawBuffer_set_cameraMatrix(self.__ptr, value.__ptr)
		return true
	elseif name == "clearColor" then
		Polycode.GPUDrawBuffer_set_clearColor(self.__ptr, value.__ptr)
		return true
	elseif name == "clearDepthBuffer" then
		Polycode.GPUDrawBuffer_set_clearDepthBuffer(self.__ptr, value)
		return true
	elseif name == "clearColorBuffer" then
		Polycode.GPUDrawBuffer_set_clearColorBuffer(self.__ptr, value)
		return true
	elseif name == "backingResolutionScale" then
		Polycode.GPUDrawBuffer_set_backingResolutionScale(self.__ptr, value.__ptr)
		return true
	elseif name == "globalMaterial" then
		Polycode.GPUDrawBuffer_set_globalMaterial(self.__ptr, value.__ptr)
		return true
	elseif name == "viewport" then
		Polycode.GPUDrawBuffer_set_viewport(self.__ptr, value.__ptr)
		return true
	end
	return false
end
function GPUDrawBuffer:GPUDrawBuffer(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.GPUDrawBuffer(unpack(arg))
	end
end

function GPUDrawBuffer:__delete()
	if self then Polycode.delete_GPUDrawBuffer(self.__ptr) end
end

File: bindings/javascript/Polycode/GPUDrawBuffer.js
function GPUDrawBuffer() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.GPUDrawBuffer()
	}
	Object.defineProperties(this, {
		'projectionMatrix': { enumerable: true, configurable: true, get: GPUDrawBuffer.prototype.__get_projectionMatrix, set: GPUDrawBuffer.prototype.__set_projectionMatrix},
		'viewMatrix': { enumerable: true, configurable: true, get: GPUDrawBuffer.prototype.__get_viewMatrix, set: GPUDrawBuffer.prototype.__set_viewMatrix},
		'cameraMatrix': { enumerable: true, configurable: true, get: GPUDrawBuffer.prototype.__get_cameraMatrix, set: GPUDrawBuffer.prototype.__set_cameraMatrix},
		'clearColor': { enumerable: true, configurable: true, get: GPUDrawBuffer.prototype.__get_clearColor, set: GPUDrawBuffer.prototype.__set_clearColor},
		'clearDepthBuffer': { enumerable: true, configurable: true, get: GPUDrawBuffer.prototype.__get_clearDepthBuffer, set: GPUDrawBuffer.prototype.__set_clearDepthBuffer},
		'clearColorBuffer': { enumerable: true, configurable: true, get: GPUDrawBuffer.prototype.__get_clearColorBuffer, set: GPUDrawBuffer.prototype.__set_clearColorBuffer},
		'backingResolutionScale': { enumerable: true, configurable: true, get: GPUDrawBuffer.prototype.__get_backingResolutionScale, set: GPUDrawBuffer.prototype.__set_backingResolutionScale},
		'viewport': { enumerable: true, configurable: true, get: GPUDrawBuffer.prototype.__get_viewport, set: GPUDrawBuffer.prototype.__set_viewport}
	})
}

GPUDrawBuffer.prototype.__get_projectionMatrix = function() {
	var retVal = new Matrix4()
	retVal.__ptr = 	Polycode.GPUDrawBuffer__get_projectionMatrix(this.__ptr)
	return retVal
}

GPUDrawBuffer.prototype.__set_projectionMatrix = function(val) {
	Polycode.GPUDrawBuffer__set_projectionMatrix(this.__ptr, val.__ptr)
}

GPUDrawBuffer.prototype.__get_viewMatrix = function() {
	var retVal = new Matrix4()
	retVal.__ptr = 	Polycode.GPUDrawBuffer__get_viewMatrix(this.__ptr)
	return retVal
}

GPUDrawBuffer.prototype.__set_viewMatrix = function(val) {
	Polycode.GPUDrawBuffer__set_viewMatrix(this.__ptr, val.__ptr)
}

GPUDrawBuffer.prototype.__get_cameraMatrix = function() {
	var retVal = new Matrix4()
	retVal.__ptr = 	Polycode.GPUDrawBuffer__get_cameraMatrix(this.__ptr)
	return retVal
}

GPUDrawBuffer.prototype.__set_cameraMatrix = function(val) {
	Polycode.GPUDrawBuffer__set_cameraMatrix(this.__ptr, val.__ptr)
}

GPUDrawBuffer.prototype.__get_clearColor = function() {
	var retVal = new Color()
	retVal.__ptr = 	Polycode.GPUDrawBuffer__get_clearColor(this.__ptr)
	return retVal
}

GPUDrawBuffer.prototype.__set_clearColor = function(val) {
	Polycode.GPUDrawBuffer__set_clearColor(this.__ptr, val.__ptr)
}

GPUDrawBuffer.prototype.__get_clearDepthBuffer = function() {
	return Polycode.GPUDrawBuffer__get_clearDepthBuffer(this.__ptr)
}

GPUDrawBuffer.prototype.__set_clearDepthBuffer = function(val) {
	Polycode.GPUDrawBuffer__set_clearDepthBuffer(this.__ptr, val)
}

GPUDrawBuffer.prototype.__get_clearColorBuffer = function() {
	return Polycode.GPUDrawBuffer__get_clearColorBuffer(this.__ptr)
}

GPUDrawBuffer.prototype.__set_clearColorBuffer = function(val) {
	Polycode.GPUDrawBuffer__set_clearColorBuffer(this.__ptr, val)
}

GPUDrawBuffer.prototype.__get_backingResolutionScale = function() {
	var retVal = new Vector2()
	retVal.__ptr = 	Polycode.GPUDrawBuffer__get_backingResolutionScale(this.__ptr)
	return retVal
}

GPUDrawBuffer.prototype.__set_backingResolutionScale = function(val) {
	Polycode.GPUDrawBuffer__set_backingResolutionScale(this.__ptr, val.__ptr)
}

GPUDrawBuffer.prototype.__get_viewport = function() {
	var retVal = new Rectangle()
	retVal.__ptr = 	Polycode.GPUDrawBuffer__get_viewport(this.__ptr)
	return retVal
}

GPUDrawBuffer.prototype.__set_viewport = function(val) {
	Polycode.GPUDrawBuffer__set_viewport(this.__ptr, val.__ptr)
}

Duktape.fin(GPUDrawBuffer.prototype, function (x) {
	if (x === GPUDrawBuffer.prototype) {
		return;
	}
	Polycode.GPUDrawBuffer__delete(x.__ptr)
})
